src/python/WMComponent/PhEDExInjector/PhEDExInjectorSubscriber.py

```diff
@@ -91,15 +95,148 @@ def algorithm(self, parameters):
 
         Run the subscription algorithm as configured
         """
+        self.deleteBlocks()
         self.subscribeDatasets()
         return
 
+    def deleteBlocks(self):
+        """
+        _deleteBlocks_
+
+        Find deletable blocks, then decide if to delete based on:
+
+        Is there an active subscription for dataset or block ?
+          If yes => set deleted=2
+          If no => next check
+
+        Has transfer to all destinations finished ?
+          If yes => request block deletion, approve request, set deleted=1
+          If no => do nothing (check again next cycle)
+
+        """
+        logging.info("Starting deleteBlocks method")
+
+        blockDict = self.findDeletableBlocks.execute(transaction = False)
+
+        if len(blockDict) == 0:
+            return
+
+        subscriptions = self.phedex.getSubscriptionMapping(*blockDict.keys())
+
+        skippableBlocks = []
+        deletableEntries = {}
+        for blockName in blockDict:
+
+            location = blockDict[blockName]['location']
+
+            # should never be triggered, better safe than sorry
+            if location.endswith('_MSS'):
+                logging.debug("Location %s for block %s is MSS, skip deletion" % (location, blockName))
+                skippableBlocks.append(blockName)
+                continue
+
+            dataset = blockDict[blockName]['dataset']
+            sites = blockDict[blockName]['sites']
+
+            if blockName in subscriptions and location in subscriptions[blockName]:
+                logging.debug("Block %s subscribed to %s, skip deletion" % (blockName, location))
+                binds = { 'DELETED' : 2,
+                          'BLOCKNAME' : blockName }
+                self.markBlocksDeleted.execute(binds, transaction = False)
+            else:
+                blockInfo = []
+                try:
+                    blockInfo = self.phedex.getReplicaInfoForBlocks(block = blockName, complete = 'y')['phedex']['block']
```

Comment:
I can't because I actually need to know that the replica is complete at the target site. For the subscriptions I wouldn't need to call PhEDEx, I could look that up inside WMBS.